node/src/components/consensus/era_supervisor.rs

```diff
@@ -606,15 +616,19 @@ where
         proto_block: ProtoBlock,
         block_context: BlockContext,
     ) -> Effects<Event<I>> {
+        if era_id.0 + BONDED_ERAS < self.era_supervisor.current_era.0 {
+            warn!(era = era_id.0, "new proto block in outdated era");
+            return Effects::new();
+        }
         let mut effects = self
             .effect_builder
             .announce_proposed_proto_block(proto_block.clone())
             .ignore();
-        // TODO: Only include _new_ accusations.
         let accusations = era_id
             .iter_bonded()
             .flat_map(|e_id| self.era(e_id).consensus.validators_with_evidence())
             .unique()
+            .filter(|pub_key| self.era(era_id).slashed.contains(pub_key))
```

Comment:
This should be a negation of `contains`, right? Otherwise it will include only validators slashed in past bonded eras. Not in the current one.

Reply:
:man_facepalming: 
I could swear I _had_ a `!` there…